app/controllers/identity_providers_controller.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+class IdentityProvidersController < ApplicationController
+  layout 'hyrax/dashboard'
+
+  before_action :ensure_admin!
+  before_action :set_identity_provider, only: %i[edit update destroy]
+
+  def index
+    @identity_providers = IdentityProvider.all
+  end
+
+  # GET /identity_providers/new
+  def new
+    add_breadcrumbs
+    @identity_provider = IdentityProvider.new
+  end
+
+  # GET /identity_providers/1/edit
+  def edit
+    add_breadcrumbs
+  end
+
+  # POST /identity_providers or /identity_providers.json
+  def create
+    @identity_provider = IdentityProvider.new(identity_provider_params)
+    respond_to do |format|
+      if @identity_provider.save
+        format.html do
+          redirect_to edit_identity_provider_url(@identity_provider),
+                      notice: "Auth provider was successfully created."
+        end
+        format.json { render :show, status: :created, location: @identity_provider }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+        format.json { render json: @identity_provider.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # PATCH/PUT /identity_providers/1 or /identity_providers/1.json
+  def update
+    respond_to do |format|
+      if @identity_provider.update(identity_provider_params)
+        format.html do
+          redirect_to edit_identity_provider_url(@identity_provider),
+                      notice: "Auth provider was successfully updated."
+        end
+        format.json { render :show, status: :ok, location: @identity_provider }
+      else
+        format.html { render :edit, status: :unprocessable_entity }
+        format.json { render json: @identity_provider.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # DELETE /identity_providers/1 or /identity_providers/1.json
+  def destroy
+    @identity_provider.destroy
+    respond_to do |format|
+      format.html { redirect_to new_identity_provider_url, notice: "Auth provider was successfully destroyed." }
+      format.json { head :no_content }
+    end
+  end
+
+  def add_breadcrumbs
+    add_breadcrumb t(:'hyrax.controls.home'), root_path
+    add_breadcrumb t(:'hyrax.dashboard.breadcrumbs.admin'), hyrax.dashboard_path
+    add_breadcrumb t(:'hyrax.admin.sidebar.configuration'), '#'
+    add_breadcrumb t(:'hyrax.admin.sidebar.identity_provider'), request.path
+  end
+
+  private
+
+    # Use callbacks to share common setup or constraints between actions.
+    def set_identity_provider
+      @identity_provider = IdentityProvider.find(params[:id])
+    end
+
+    def ensure_admin!
+      authorize! :read, :admin_dashboard
```

Comment:
This was a bit surprising...I would've assumed greater privileges than reading the admin dashboard.